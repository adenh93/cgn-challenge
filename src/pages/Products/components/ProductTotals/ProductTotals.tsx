import React, { SFC } from "react";
import ProductTotalsItem from "../ProductTotalsItem";

interface ProductTotalsProps {
  total: number;
  productCount: number;
  isWaiting: boolean;
}

const ProductTotals: SFC<ProductTotalsProps> = ({
  total,
  productCount,
  isWaiting
}) => (
  <>
    <ProductTotalsItem
      label="Total Price"
      amount={total}
      isWaiting={isWaiting}
    />
    <ProductTotalsItem
      label="Average Price"
      amount={productCount ? total / productCount : 0}
      isWaiting={isWaiting}
    />
  </>
);

export default ProductTotals;
