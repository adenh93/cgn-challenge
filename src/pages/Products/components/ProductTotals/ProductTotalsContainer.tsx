import React, { SFC } from "react";
import { ProductsResponse } from "../../../../types";
import { connect } from "react-redux";
import { ApplicationState } from "../../../../store";
import ProductTotals from "./ProductTotals";

interface ProductTotalsContainerProps {
  productsResponse: ProductsResponse;
  apiCallsInProgress: number;
}

const ProductTotalsContainer: SFC<ProductTotalsContainerProps> = ({
  productsResponse,
  apiCallsInProgress
}) => {
  const { results: products } = productsResponse;

  const getTotal = () => {
    if (!products.length) return 0;

    const sum = products
      .map(({ price }) => price)
      .reduce((prev, next) => prev + next);

    return sum;
  };

  return (
    <>
      <ProductTotals
        total={getTotal()}
        isWaiting={apiCallsInProgress > 0}
        productCount={products.length}
      />
    </>
  );
};

const mapStateToProps = ({
  productState,
  apiStatusState
}: ApplicationState) => {
  return {
    productsResponse: productState.productsResponse,
    apiCallsInProgress: apiStatusState.apiCallsInProgress
  };
};

export default connect(mapStateToProps)(ProductTotalsContainer);
