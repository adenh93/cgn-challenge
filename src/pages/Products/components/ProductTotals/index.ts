import ProductTotalsContainer from "./ProductTotalsContainer";
export default ProductTotalsContainer;
