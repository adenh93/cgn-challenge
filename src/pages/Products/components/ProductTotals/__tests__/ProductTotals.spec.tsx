import * as React from "react";
import { create } from "react-test-renderer";
import ProductTotals from "../ProductTotals";

describe("ProductTotals Component", () => {
  test("it matches the snapshot", () => {
    const component = create(
      <ProductTotals total={500} productCount={5} isWaiting={false} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
