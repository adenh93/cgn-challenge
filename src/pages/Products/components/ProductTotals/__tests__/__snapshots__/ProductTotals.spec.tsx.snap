// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTotals Component it matches the snapshot 1`] = `
Array [
  <h6
    className="MuiTypography-root makeStyles-root-1 MuiTypography-body1"
  >
    Total Price
    :
     
    <small
      className="makeStyles-small-2"
    >
      $
      500.00
    </small>
  </h6>,
  <h6
    className="MuiTypography-root makeStyles-root-1 MuiTypography-body1"
  >
    Average Price
    :
     
    <small
      className="makeStyles-small-2"
    >
      $
      100.00
    </small>
  </h6>,
]
`;
